aiidalab_widgets_base/viewers.py

```diff
@@ -222,7 +228,91 @@ def _download_tab(self):
+        # 4. Render a high quality image
+        self.render_btn = ipw.Button(description="Render", icon='fa-paint-brush')
+        self.render_btn.on_click(self._render_structure)
+        self.render_box = ipw.VBox(children=[ipw.Label("Render a image with POVRAY:"), self.render_btn])
+
+        return ipw.VBox([self.download_box, self.screenshot_box, self.render_box])
+
+    def _render_structure(self, change=None):
+        """Render the structure with POVRAY."""
+
+        if not isinstance(self.structure, Atoms):
+            return 
+
+        self.render_btn.disabled = True 
+        omat = np.array(self._viewer._camera_orientation)
+        omat = omat.reshape(4, 4)
+        omat = np.transpose(omat)
+
+        zfactor = norm(omat[0, 0:3])
+        omat[0:3, 0:3] = omat[0:3, 0:3]/zfactor
+
+        bb = deepcopy(self.structure)
+
+        for i in bb:
+            ixyz = np.array([i.x, i.y, i.z])
+            ixyz = ixyz + omat[0:3, 3];
+            ixyz = omat[0:3, 0:3].dot(ixyz)
+            i.x = -ixyz[0]
+            i.y =  ixyz[1]
+            i.z =  ixyz[2]
+        
+        vertices = [];
+    
+        cell = bb.get_cell()
+        vertices.append(np.array([0, 0, 0]))
+        vertices.extend(cell)
+        vertices.extend([cell[0]+cell[1], cell[0]+cell[2], cell[1]+cell[2], cell[0]+cell[1]+cell[2]])
+
+        for n, i in enumerate(vertices):
+            ixyz = omat[0:3, 0:3].dot(i + omat[0:3, 3])
+            vertices[n] = np.array([-ixyz[0], ixyz[1], ixyz[2]])
+    
+        camera = Camera('location', [0, 0, -zfactor/1.5], 'look_at', [0.0, 0.0, 0.0])
+        light = LightSource([0, 0, -100.0], 'color',  [1.5, 1.5, 1.5])
+
+        spheres = [Sphere( [i.x, i.y, i.z], Radius[i.symbol], 
+                            Texture(Pigment( 'color', np.array(Colors[i.symbol]))), 
+                                Finish('phong', 0.9,'reflection', 0.05)) for i in bb]
+
+        bonds = []
+        for x, i in enumerate(bb):
+            for j in bb[x+1:]:
+                v1 = np.array([i.x, i.y, i.z])
+                v2 = np.array([j.x, j.y, j.z])
+
+                if i.symbol == 'H' and j.symbol == 'H':
+                    continue
+                
+                if norm(v1-v2) < 1.4*(Radius[i.symbol] + Radius[j.symbol]):
+                    midi = v1 + (v2-v1)*Radius[i.symbol]/(Radius[i.symbol] + Radius[j.symbol]);
+                    bond = Cylinder(v1, midi, 0.2, Pigment('color', np.array(Colors[i.symbol])),
+                                    Finish('phong', 0.8,'reflection', 0.05))
+                    bonds.append(bond)
+                    bond = Cylinder(v2, midi, 0.2, Pigment('color', np.array(Colors[j.symbol])),
+                                    Finish('phong', 0.8,'reflection', 0.05))
+                    bonds.append(bond)
+                
+        edges = []
+        for x, i in enumerate(vertices):
+            for y, j in enumerate(vertices):
+                if y > x:
```

Comment:
Can you use here the same trick as above? Something like:
```python
        for x, i in enumerate(vertices):
            for j in vertices[x+1:]:
```

Reply:
Yes, it is fixed.